public/app/features/explore/ExploreActions.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC, useEffect, useState } from 'react';
+import { useRegisterActions, useKBar, Action, Priority } from 'kbar';
+import { useDispatch, useSelector } from 'react-redux';
+import { ExploreId } from 'app/types';
+import { splitOpen, splitClose } from './state/main';
+import { isSplit } from './state/selectors';
+import { runQueries } from './state/query';
+
+interface Props {
+  exploreIdLeft: ExploreId;
+  exploreIdRight?: ExploreId;
+}
+
+export const ExploreActions: FC<Props> = ({ exploreIdLeft, exploreIdRight }: Props) => {
+  const [actions, setActions] = useState<Action[]>([]);
+  const { query } = useKBar();
+  const dispatch = useDispatch();
+  const splitted = useSelector(isSplit);
+
+  useEffect(() => {
+    const exploreSection = {
+      name: 'Explore',
+      priority: Priority.HIGH + 1,
+    };
+
+    const actionsArr: Action[] = [];
+
+    if (splitted) {
+      actionsArr.push({
+        id: 'explore/run-query-left',
+        name: 'Run Query (Left)',
+        keywords: 'query left',
+        perform: () => {
+          dispatch(runQueries(exploreIdLeft));
+        },
+        section: exploreSection,
+      });
+      if (exploreIdRight) {
+        // we should always have the right exploreId if split
+        actionsArr.push({
+          id: 'explore/run-query-right',
+          name: 'Run Query (Right)',
+          keywords: 'query right',
+          perform: () => {
+            dispatch(runQueries(exploreIdRight));
+          },
+          section: exploreSection,
+        });
+        actionsArr.push({
+          id: 'explore/split-view-close-left',
+          name: 'Close split view left',
+          keywords: 'split',
+          perform: () => {
+            dispatch(splitClose(exploreIdRight));
+          },
+          section: exploreSection,
+        });
+        actionsArr.push({
+          id: 'explore/split-view-close-right',
+          name: 'Close split view right',
+          keywords: 'split',
+          perform: () => {
+            dispatch(splitClose(exploreIdRight));
+          },
+          section: exploreSection,
+        });
+      }
+    } else {
+      actionsArr.push({
+        id: 'explore/run-query',
+        name: 'Run Query',
+        keywords: 'query',
+        perform: () => {
+          dispatch(runQueries(exploreIdLeft));
+        },
+        section: exploreSection,
+      });
+      actionsArr.push({
+        id: 'explore/split-view-open',
+        name: 'Open split view',
+        keywords: 'split',
+        perform: () => {
+          dispatch(splitOpen());
+        },
+        section: exploreSection,
+      });
+    }
+    setActions(actionsArr);
+  }, [exploreIdLeft, exploreIdRight, splitted, query, dispatch]);
+
+  useRegisterActions(!query ? [] : actions, [actions, query]);
```

Comment:
according to TS, `query` is non-nullable. is it correct?

```suggestion
  useRegisterActions(actions, [actions]);
```

Reply:
Yep, this is a holdover from one of the previous issues I was having with the library but it works fine without checking for query now :D